src/binaryen-c.cpp

```diff
@@ -3122,6 +3145,33 @@ void BinaryenModuleOptimize(BinaryenModuleRef module) {
   passRunner.run();
 }
 
+BinaryenFeatureFlags BinaryenModuleGetFeatures(BinaryenModuleRef module) {
+  if (tracing) {
+    std::cout << "  BinaryenModuleGetFeatures(the_module);\n";
+  }
+
+  Module* wasm = (Module*)module;
+  BinaryenFeatureFlags featureFlags = 0;
+  wasm->features.iterFeatures(
+    [&featureFlags](FeatureSet::Feature flag) { featureFlags |= flag; });
+  return featureFlags;
+}
+
+void BinaryenModuleSetFeatures(BinaryenModuleRef module,
+                               BinaryenFeatureFlags featureFlags) {
+  if (tracing) {
+    std::cout << "  BinaryenModuleSetFeatures(the_module, " << featureFlags
+              << ");\n";
+  }
+
+  Module* wasm = (Module*)module;
+  for (BinaryenFeatureFlags flag = 1; flag < FeatureSet::Feature::All;
+       flag <<= 1) {
```

Comment:
This might be more elegant using iterFeatures over a featureset initialized with All.